selfdrive/car/hyundai/carstate.py

```diff
@@ -285,16 +292,20 @@ def get_can_parser(self, CP):
     else:
       messages.append(("LVR12", 100))
 
-    return CANParser(DBC[CP.carFingerprint]["pt"], messages, 0)
+    bus = CanBus(CP).ECAN if CP.flags & HyundaiFlags.CAN_CANFD.value else 0
+    return CANParser(DBC[CP.carFingerprint]["pt"], messages, bus)
 
   @staticmethod
   def get_cam_can_parser(CP):
     if CP.carFingerprint in CANFD_CAR:
       return CarState.get_cam_can_parser_canfd(CP)
 
-    messages = [
-      ("LKAS11", 100)
-    ]
+    messages = []
+
+    if CP.flags & HyundaiFlags.CAN_CANFD:
+      messages.append(("CAM_0x2a4", 20))
```

Comment:
here you assume we can only use `CAM_0x2a4` if hybrid CAN FD, but above you use the `get_hda2_lfa_block_msg` helper function. i'd make this explicit, one way or the other (probably only using 0x2a4 since we've never seen a hybrid CAN FD using the other one.